src/lib/kdf/hkdf/hkdf.cpp

```diff
@@ -46,42 +44,31 @@ std::string HKDF_Extract::name() const {
    return fmt("HKDF-Extract({})", m_prf->name());
 }
 
-void HKDF_Extract::kdf(uint8_t key[],
-                       size_t key_len,
-                       const uint8_t secret[],
-                       size_t secret_len,
-                       const uint8_t salt[],
-                       size_t salt_len,
-                       const uint8_t /*label*/[],
-                       size_t label_len) const {
-   if(key_len == 0) {
+void HKDF_Extract::perform_kdf(std::span<uint8_t> key,
+                               std::span<const uint8_t> secret,
+                               std::span<const uint8_t> salt,
+                               std::span<const uint8_t> label) const {
+   if(key.empty()) {
```

Comment:
We should probably even still check that the label was empty

Reply:
I do, in the `BOTAN_ARG_CHECK` below. That replaced the previous explicit `if(label_len > 0) throw Invalid_Argument`. Or am I missing something?

Reply:
My point was that here if the key is empty we return immediately, prior to checking the label. So a request for HKDF of length zero with label "Foo" returns an empty string, even though HKDF isn't defined for non-empty labels.

Reply:
Fair. I moved the arg checks up. Also in HKDF expand.